### PR TITLE
[FEAT] add bank account

### PR DESCRIPTION
**Pull request by nonya123456**
create shop body:
```
{
    "name": "126123112fsdfsdf161",
    "image": null,
    "description": "wef",
    "location": "wefwe",
    "contact": "wef",
    "owner_emails": [
        "nonya579@gmail.com"
    ],
    "bank_account": {
        "name": "test2 test2",
        "number": "123456789",
        "brand": "kbank",
        "type": "individual"
    }
}
```

### CODE REVIEW

File: apigateway/internal/v1router/types/shop.go [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-	Name   string `json:"name"`
-	Type   string `json:"type"`
-	Number string `json:"number"`
-	Brand  string `json:"brand"`
+import (
+	"github.com/GrabItYourself/giys-backend/payment/pkg/paymentproto"
+	"github.com/GrabItYourself/giys-backend/shop/pkg/shopproto"
+)
+
+type CreateShopWithBankAccountRequest struct {
```
Comment (bombnp) [inline apigateway/internal/v1router/types/shop.go]: why not just add bank account fields inside the CreateShopRequest proto itself? so we don't need custom type for this

File: apigateway/internal/v1router/handler/shop.go [hunk 2]
```diff
@@ -64,10 +64,19 @@ func (h *Handler) HandleCreateShop(c *fiber.Ctx, shop *shopproto.CreateShopReque
...
 	}
+
+	_, err = h.Grpc.Payment.RegisterRecipient(ctx, &paymentproto.RegisterRecipientRequest{
+		ShopId:      res.Shop.Id,
+		BankAccount: &shop.BankAccount,
+	})
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, errors.Wrap(err, "Failed to request GRPC payment").Error())
+	}
+
```
Comment (bombnp) [inline apigateway/internal/v1router/handler/shop.go]: should we move this rpc call to CreateShop method in ShopService instead?

**Review by bombnp — CHANGES_REQUESTED** [review]
move register recipient logic to ShopService instead? I don't see why not

**Review by saenyakorn — COMMENTED** [review]
ถ้าไม่ยากเกินไป ฝากทำ PUT shop ด้วยได้มะ เรื่อง edit bank account

**Review by bombnp — APPROVED** [review]

**Pull request merged**